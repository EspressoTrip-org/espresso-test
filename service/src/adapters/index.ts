export * as policy_cache from './policy-cache';
