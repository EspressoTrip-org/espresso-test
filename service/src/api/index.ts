export * as organization from './organizations';
export * as compilation from './compilation';
export * as policies from './policies';
export * as tokens from './tokens';
export * as roles from './roles';
export * as users from './users';
