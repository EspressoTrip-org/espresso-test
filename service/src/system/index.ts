import { InterServiceClient } from '@journeyapps-platform/sdk-inter-service';
import * as entity_resolver from '@journeyapps-platform/recon-entity-resolver';
import * as invalidator from '@journeyapps-platform/recon-event-invalidator';
import * as micro from '@journeyapps-platform/micro';
import * as adapters from '../adapters';
import * as kafka_utils from './kafka';
import * as resolver from '../recon';
import { MongoDB } from './mongo';
import * as kafka from 'kafkajs';
import ENV from '../env';

export type CreateSystemParams = {
  mocks: string[];
  accounts: {
    endpoint: string;
    client_id: string;
    client_secret: string;
  };
  mongo: micro.mongo.MongoDBAdapterOptions;
  kafka: micro.kafka.KafkaConnectionOptions;
  node_env: string;
};

export class System extends micro.system.MicroSystem {
  context: CreateSystemParams;

  mongo: MongoDB;
  kafka: micro.kafka.KafkaClientProvisioner;
  producer: kafka.Producer;
  resolver: entity_resolver.EntityResolver;

  policy_cache: adapters.policy_cache.PolicyCache;

  constructor(params: CreateSystemParams) {
    super();

    this.context = params;

    const inter_service = new InterServiceClient(params.accounts);

    this.mongo = this.withLifecycle(new MongoDB(params.mongo), {
      start(component) {
        return component.connect();
      },
      stop(component) {
        return component.close();
      }
    });

    this.kafka = this.withLifecycle(micro.kafka.createClientProvisioner(params.kafka), {
      start(component) {
        return kafka_utils.ensureTopics({
          kafka: component(),
          production: params.node_env === 'production'
        });
      }
    });

    this.producer = this.withLifecycle(
      this.kafka('producer').producer({
        createPartitioner: kafka.Partitioners.LegacyPartitioner
      }),
      {
        start(component) {
          return component.connect();
        },
        stop(component) {
          return component.disconnect();
        }
      }
    );

    this.resolver = this.withLifecycle(
      resolver.createResolver(
        this.mongo,
        params.mocks.includes('recon')
          ? undefined
          : {
              external: {
                endpoint: params.accounts.endpoint,
                client: inter_service
              }
            }
      ),
      {}
    );

    this.withLifecycle(
      invalidator.createEventInvalidator({
        resolver: this.resolver,
        kafka: this.kafka('invalidator'),
        modules: [invalidator.resource_operation_events_module, invalidator.cardinal_module]
      }),
      {
        start: async (invalidator) => {
          await invalidator.start();
          await this.resolver.init();
        },
        stop(invalidator) {
          return invalidator.stop();
        }
      }
    );

    this.policy_cache = adapters.policy_cache.createInMemoryPolicyCache();
  }

  static fromENV(env: typeof ENV) {
    return new System({
      mocks: env.MOCK,
      accounts: {
        endpoint: env.JWT_ISSUER,
        client_id: env.ACCOUNTS_CLIENT_ID,
        client_secret: env.ACCOUNTS_CLIENT_SECRET
      },
      mongo: {
        uri: env.MONGO_URI,
        database: env.MONGO_DATABASE,
        username: env.MONGO_USERNAME,
        password: env.MONGO_PASSWORD
      },
      kafka: {
        brokers: env.KAFKA_BROKERS,
        username: env.KAFKA_USERNAME,
        password: env.KAFKA_PASSWORD,
        mechanism: env.KAFKA_MECHANISM as any
      },
      node_env: env.NODE_ENV
    });
  }
}
