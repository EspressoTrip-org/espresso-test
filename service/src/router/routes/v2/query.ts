import * as cardinal from '@journeyapps-platform/types-cardinal';
import * as micro from '@journeyapps-platform/micro';
import { Router } from '../../router';
import * as errors from '../../../errors';
import * as api from '../../../api';
import { CardinalModel, TokenModelAction, UserModelAction } from '@journeyapps-platform/cardinal-catalog';

export const policies_for_user = Router.post(cardinal.ROUTES_V2.QUERY_POLICIES_FOR_USER, {
  validator: micro.schema.createTsCodecValidator(cardinal.QueryPoliciesForUser),
  authorize: async ({ params, context }) => {
    return context.viewer.can(UserModelAction.READ, CardinalModel.USER, params.user_id);
  },
  handler: ({ params, context }) => {
    return api.policies.getPoliciesForUser(context.system, {
      user_id: params.user_id,
      policies: context.viewer.policies
    });
  }
});

export const policies_for_token = Router.post(cardinal.ROUTES_V2.QUERY_POLICIES_FOR_TOKEN, {
  validator: micro.schema.createTsCodecValidator(cardinal.QueryPoliciesForToken),
  authorize: async ({ params, context }) => {
    const token = await api.tokens.getTokenByValue(context.system.mongo, params.token);
    if (!token) {
      throw new errors.Error404('token', '<redacted>');
    }
    return context.viewer.can(TokenModelAction.READ, CardinalModel.TOKEN, token.id);
  },
  handler: async ({ params, context }) => {
    const token = await api.tokens.getTokenByValue(context.system.mongo, params.token);
    if (!token) {
      throw new errors.Error404('token', '<redacted>');
    }
    return api.policies.getPoliciesForToken(context.system, {
      token_id: token.id,
      policies: context.viewer.policies
    });
  }
});

export const routes = [policies_for_user, policies_for_token];
