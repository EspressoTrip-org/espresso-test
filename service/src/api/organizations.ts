import * as cardinal from '@journeyapps-platform/types-cardinal';
import * as micro from '@journeyapps-platform/micro';
import * as producer from './producer';
import * as policies from './policies';
import * as tokens from './tokens';
import { System } from '../system';
import * as roles from './roles';
import * as users from './users';
import * as auth from '../auth';
import * as bson from 'bson';

type PolicyProvisionParams = {
  org_id: string;
};

export const provisionStandardPoliciesAndRolesForOrg = async (system: System, params: PolicyProvisionParams) => {
  const actor: cardinal.SystemActor = {
    type: cardinal.ActorType.System
  };

  const generated_policies = auth.createDefaultPoliciesForOrg(params.org_id);
  const new_policies = await Promise.all(
    Object.values(generated_policies).map((policy) => {
      return policies.createPolicy(system, {
        actor: actor,
        policy: {
          ...policy,
          managed: true
        }
      });
    })
  );

  const policy_map = new_policies.reduce((acc: auth.CreateDefaultRolesPolicyMap, policy) => {
    acc[policy.name as auth.MANAGED_POLICY] = policy.id;
    return acc;
  }, {});

  const generated_roles = auth.createDefaultRolesForOrg(params.org_id, policy_map);
  return await Promise.all(
    generated_roles.map((role) => {
      return roles.createRole(system, {
        actor,
        role: {
          ...role,
          managed: true
        }
      });
    })
  );
};

type EmitInvalidationEventParams = {
  org: micro.db.SerializedId & cardinal.Organization;
};

/**
 * Emit user assignment changed events which act as cache invalidation events.
 */
export const emitInvalidationEvents = async (system: System, params: EmitInvalidationEventParams) => {
  const org_users = await users.listUsersForOrganization(system.mongo, {
    org_id: params.org.id
  });

  await producer.produceAuthEvents(system, {
    actor: {
      type: cardinal.ActorType.System
    },
    events: org_users.map((user) => {
      return {
        type: cardinal.UserAuthEventType.USER_ASSIGNMENTS_CHANGED,
        payload: user
      };
    })
  });
};

type UpsertOrganizationParams = {
  org: micro.db.SerializedId & cardinal.Organization;
  created_by?: string;
};

export const upsertOrganization = async (system: System, params: UpsertOrganizationParams) => {
  const existing = await system.mongo.organizations.findOne(micro.mongo.getById(params.org.id));

  await system.mongo.organizations.updateOne(
    {
      _id: new bson.ObjectId(params.org.id)
    },
    {
      $set: {
        name: params.org.name,
        locked: params.org.locked
      }
    },
    { upsert: true }
  );

  if (existing) {
    const locked_changed = params.org.locked !== existing.locked;
    if (locked_changed) {
      await emitInvalidationEvents(system, params);
    }

    return;
  }

  const roles = await provisionStandardPoliciesAndRolesForOrg(system, {
    org_id: params.org.id
  });

  // If there is a created_by field on the event then assign the Owner role to the user who
  // created this organization
  if (!params.created_by) {
    return;
  }

  try {
    const user = await users.getUser(system.mongo, params.created_by);
    const owner_role = roles.find((role) => role.name === cardinal.MANAGED_ROLE.OWNER);
    if (!owner_role) {
      return;
    }

    await users.updateUserAssignment(system, {
      actor: {
        type: cardinal.ActorType.System
      },
      assignment: 'role_ids',
      assignment_id: owner_role.id,
      id: user.id,
      op: 'add'
    });
  } catch (e) {
    micro.logger.error(`user ${params.created_by} (creator of org) could not be found`);
  }
};

export const deleteOrganization = async (system: System, params: { id: string }) => {
  await system.mongo.organizations.deleteOne({
    _id: new bson.ObjectId(params.id)
  });

  const all_policies = await policies.listScopedPoliciesForOrganization(system.mongo, {
    org_id: params.id
  });
  const all_roles = await roles.listRolesByOrganization(system.mongo, {
    org_id: params.id
  });
  const all_tokens = await tokens.listScopedTokensForOrganization(system.mongo, {
    org_id: params.id
  });

  // This is a pretty inefficient way to clean up but org deletion will be very in-frequent
  // and as such this is a nice way to re-use already tested logic to do the cleanup

  for (const token of all_tokens) {
    await tokens.deleteToken(system, {
      actor: {
        type: cardinal.ActorType.System
      },
      id: token.id
    });
  }

  for (const role of all_roles) {
    await roles.deleteRole(system, {
      actor: {
        type: cardinal.ActorType.System
      },
      id: role.id
    });
  }

  for (const policy of all_policies) {
    await policies.deletePolicy(system, {
      actor: {
        type: cardinal.ActorType.System
      },
      id: policy.id
    });
  }
};
