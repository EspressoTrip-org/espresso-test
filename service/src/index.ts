import '@journeyapps-platform/micro/register';

import * as micro from '@journeyapps-platform/micro';
import * as routes from './router/routes/index';
import { Router } from './router/router';
import * as consumers from './consumers/index';
import { System } from './system/index';
import * as auth from './auth/index';
import fastify from 'fastify';
import env from './env';

const server = fastify();

const system = System.fromENV(env);
const jwt_decoder = auth.createJWTDecoder(system, {
  issuer_url: env.JWT_ISSUER,
  verify: !env.DEV_MODE_DO_NOT_ENABLE_IN_PRODUCTION_OR_YOU_WILL_BE_FIRED
});

const token_decoder = auth.createTokenDecoder(system);
const dev_decoder = auth.createDevDecoder();

const decoders = [jwt_decoder, token_decoder];
if (env.DEV_MODE_DO_NOT_ENABLE_IN_PRODUCTION_OR_YOU_WILL_BE_FIRED) {
  decoders.push(dev_decoder);
}

server.register(
  Router.plugin({
    routes: [...routes.v1_routes, ...routes.v2_routes, ...micro.router.createProbeRoutes()],
    contextProvider: async (payload) => {
      return {
        system: system,
        viewer: await auth.createViewer({
          token: payload.request.headers['authorization'] || '',
          system,
          decoders
        }),
        sha: env.SHA
      };
    }
  })
);

(async () => {
  await system.start();

  await micro.fastify.startServer(server, env.PORT);
  micro.logger.info(`HTTP server running on port ${env.PORT}`);
  await micro.signals.getSystemProbe().ready();

  const [resource_events_consumer, cache_invalidator] = await Promise.all([
    consumers.createResourceOperationEventsConsumer(system),
    consumers.createCacheInvalidator(system)
  ]);

  micro.signals.getTerminationHandler()?.handleTerminationSignal(async () => {
    await Promise.all([resource_events_consumer.disconnect(), cache_invalidator.disconnect()]);
  });
})().catch((err) => {
  micro.logger.error('Something went wrong while booting', err);
  process.exit(1);
});
