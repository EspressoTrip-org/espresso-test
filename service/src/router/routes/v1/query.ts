// transitive dependencies
import type * as micro from '@journeyapps-platform/micro';

import { Router } from '../../router';
import * as v2 from '../v2/query';

export const policies_for_user = Router.get('/api/v1/users/:user_id/get-policies', {
  validator: v2.policies_for_user.validator,
  authorize: v2.policies_for_user.authorize,
  handler: v2.policies_for_user.handler
});

export const policies_for_token = Router.post('/api/v1/tokens/get-policies', {
  validator: v2.policies_for_token.validator,
  authorize: v2.policies_for_token.authorize,
  handler: v2.policies_for_token.handler
});

export const query_routes = [policies_for_user, policies_for_token];
