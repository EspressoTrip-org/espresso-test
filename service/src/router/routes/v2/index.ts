// transitive dependency
import type * as micro from '@journeyapps-platform/micro';

import * as assignments from './assignments';
import * as policies from './policies';
import * as tokens from './tokens';
import * as roles from './roles';
import * as query from './query';

export const v2_routes = [
  ...assignments.routes,
  ...policies.routes,
  ...roles.routes,
  ...tokens.routes,
  ...query.routes
];
