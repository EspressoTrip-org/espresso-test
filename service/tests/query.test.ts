import { beforeAll, beforeEach, afterAll, describe, test, expect } from 'vitest';

import * as cardinal from '@journeyapps-platform/types-cardinal';
import * as routes from '../src/router/routes/v2/query';
import { System } from '../src/system';
import * as api from '../src/api';
import * as mocks from './mocks';
import * as bson from 'bson';
import { CardinalModel, PolicyModelAction } from '@journeyapps-platform/cardinal-catalog';

let system: System;

const org_id = '5f2ab36c60d2294576efa646';

beforeAll(async () => {
  system = await mocks.createMockedSystem();
  await system.start();

  await api.organization.upsertOrganization(system, {
    org: {
      id: org_id,
      name: 'test org'
    }
  });
});

afterAll(async () => {
  await system.stop();
});

beforeEach(async () => {
  // @ts-ignore
  await system.mongo.policies.deleteMany({});
  await system.mongo.users.deleteMany({});
});

describe('policy queries', () => {
  const policy_id = new bson.ObjectId('5e6f5519ede3631176831f4c');
  const now = new Date('2020-03-20T12:31:48.370Z');

  const policy: cardinal.PolicyResource = {
    _id: policy_id,
    name: 'Policy',
    description: 'Some Policy',
    org_id: org_id,
    statements: [],
    created_at: now,
    updated_at: now
  };

  const policies: cardinal.Policy[] = [
    {
      statements: [
        {
          actions: [PolicyModelAction.READ],
          resources: [
            {
              scope: '*',
              selector: {
                model: CardinalModel.POLICY,
                id: '*'
              }
            }
          ]
        },
        {
          actions: [PolicyModelAction.READ],
          resources: [
            {
              selector: {
                model: CardinalModel.POLICY,
                id: '*'
              }
            }
          ]
        }
      ]
    }
  ];

  test('return policies for user', async () => {
    await system.mongo.policies.insertOne(policy);

    const user = await system.mongo.users.insertOne({
      _id: new bson.ObjectId(),

      email: 'test-user@journeyapps.com',
      policy_ids: [policy_id.toHexString()],
      role_ids: [],
      org_id: org_id
    });

    await system.resolver.init();

    const res = await routes.policies_for_user.handler(
      mocks.createMockedPayload({
        system,
        policies,
        params: {
          user_id: user.insertedId.toHexString()
        }
      })
    );
    expect(res).toMatchSnapshot();
  });

  test('policy statements should be interpolated on request', async () => {
    await system.mongo.policies.insertOne({
      ...policy,
      statements: [
        {
          actions: ['read'],
          resources: [
            {
              scope: policy.org_id,
              selector: {
                model: 'draft',
                id: '*',
                parents: [
                  {
                    model: 'user',
                    id: '$actor.id'
                  }
                ]
              }
            }
          ]
        }
      ]
    });

    const user_id = new bson.ObjectId('5e6f5519ede3631176832f4a');
    await system.mongo.users.insertOne({
      _id: user_id,

      email: 'test-user@journeyapps.com',
      policy_ids: [policy_id.toHexString()],
      role_ids: [],
      org_id: org_id
    });

    await system.resolver.init();

    const res = await routes.policies_for_user.handler(
      mocks.createMockedPayload({
        system,
        policies,
        params: {
          user_id: user_id.toHexString()
        }
      })
    );
    expect(res).toMatchSnapshot();
  });

  test('return policies for user, filtering out those for locked orgs', async () => {
    // Fixtures we want returned
    await system.mongo.policies.insertOne(policy);

    // Fixtures we want filtered out.
    const locked_org_id = '5f2ab78a678d1d478d17e5cf';
    await api.organization.upsertOrganization(system, {
      org: {
        id: locked_org_id,
        locked: true,
        name: 'Locked Org'
      }
    });

    const locked_org_policy_id = new bson.ObjectId('5f3d63ce7a73979876d9c4d9');
    await system.mongo.policies.insertOne({
      _id: locked_org_policy_id,
      name: 'Policy',
      description: 'Some Policy for Locked Org',
      org_id: locked_org_id,
      statements: [],
      created_at: now,
      updated_at: now
    });

    const unscoped_policy_id = new bson.ObjectId('5f3d63ce7a73979876d9c4c7');
    await system.mongo.policies.insertOne({
      _id: unscoped_policy_id,
      name: 'Unscoped Policy',
      description: 'An unscoped policy',
      statements: [],
      created_at: now,
      updated_at: now
    });

    const user = await system.mongo.users.insertOne({
      _id: new bson.ObjectId(),

      email: 'test-user@journeyapps.com',
      policy_ids: [policy_id.toHexString(), locked_org_policy_id.toHexString(), unscoped_policy_id.toHexString()],
      role_ids: [],
      org_id: org_id
    });

    await system.resolver.init();

    const res = await routes.policies_for_user.handler(
      mocks.createMockedPayload({
        system,
        policies,
        params: {
          user_id: user.insertedId.toHexString()
        }
      })
    );
    expect(res).toMatchSnapshot();
  });

  test('return locked policy for user when org is locked', async () => {
    await system.mongo.policies.insertOne(policy);

    const org_id = '5f2ab78a678d1d478d17e5cf';
    await api.organization.upsertOrganization(system, {
      org: {
        id: org_id,
        locked: true,
        name: 'Locked Org'
      }
    });
    const user = await system.mongo.users.insertOne({
      _id: new bson.ObjectId(),

      email: 'test-user@journeyapps.com',
      policy_ids: [policy_id.toHexString()],
      role_ids: [],
      org_id: org_id
    });

    await system.resolver.init();

    const res = await routes.policies_for_user.handler(
      mocks.createMockedPayload({
        system,
        policies,
        params: {
          user_id: user.insertedId.toHexString()
        }
      })
    );

    expect(res).toMatchSnapshot();
  });

  test('return no policies for user due to permissions', async () => {
    await system.mongo.policies.insertOne(policy);

    const user = await system.mongo.users.insertOne({
      _id: new bson.ObjectId(),

      email: 'test-user@journeyapps.com',
      policy_ids: [policy_id.toHexString()],
      role_ids: [],
      org_id: org_id
    });

    await system.resolver.init();

    const res = await routes.policies_for_user.handler(
      mocks.createMockedPayload({
        system,
        policies: [],
        params: {
          user_id: user.insertedId.toHexString()
        }
      })
    );
    expect(res).toMatchSnapshot();
  });

  test('return policies for token', async () => {
    await system.mongo.policies.insertOne(policy);
    const token = await api.tokens.createToken(system, {
      actor: cardinal.SYSTEM_ACTOR,
      token: {
        description: 'test token',
        org_id: policy.org_id,
        policy_ids: [policy_id.toHexString()]
      }
    });

    await system.resolver.init();

    const res = await routes.policies_for_token.handler(
      mocks.createMockedPayload({
        system,
        policies,
        params: {
          token: token.value
        }
      })
    );
    expect(res).toMatchSnapshot();
  });

  test('return no policies for token belonging to locked organizations', async () => {
    await system.mongo.policies.insertOne(policy);

    const org_id = '5f2ab78a678d1d478d17e1a6';
    await api.organization.upsertOrganization(system, {
      org: {
        id: org_id,
        locked: true,
        name: 'Locked Org'
      }
    });

    const token = await api.tokens.createToken(system, {
      actor: cardinal.SYSTEM_ACTOR,
      token: {
        description: 'test token',
        org_id: org_id,
        policy_ids: [policy_id.toHexString()]
      }
    });

    await system.resolver.init();

    const res = await routes.policies_for_token.handler(
      mocks.createMockedPayload({
        system,
        policies,
        params: {
          token: token.value
        }
      })
    );
    expect(res.length).toBe(0);
  });
});
