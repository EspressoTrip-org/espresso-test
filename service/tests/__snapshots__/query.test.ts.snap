// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`policy queries > policy statements should be interpolated on request 1`] = `
[
  {
    "id": "5e6f5519ede3631176831f4c",
    "org_id": "5f2ab36c60d2294576efa646",
    "statements": [
      {
        "actions": [
          "read",
        ],
        "resources": [
          {
            "scope": "5f2ab36c60d2294576efa646",
            "selector": {
              "id": "*",
              "model": "draft",
              "parents": [
                {
                  "id": "5e6f5519ede3631176832f4a",
                  "model": "user",
                },
              ],
            },
          },
        ],
      },
    ],
  },
]
`;

exports[`policy queries > return locked policy for user when org is locked 1`] = `
[
  {
    "description": "Limited set of permissions for locked organizations",
    "name": "locked",
    "statements": [
      {
        "actions": [
          "read",
        ],
        "resources": [
          {
            "selector": {
              "id": "5f2ab78a678d1d478d17e5cf",
              "model": "organization",
            },
          },
          {
            "scope": "5f2ab78a678d1d478d17e5cf",
            "selector": {
              "id": "*",
              "model": "app",
            },
          },
        ],
      },
    ],
  },
]
`;

exports[`policy queries > return no policies for user due to permissions 1`] = `[]`;

exports[`policy queries > return policies for token 1`] = `
[
  {
    "id": "5e6f5519ede3631176831f4c",
    "org_id": "5f2ab36c60d2294576efa646",
    "statements": [],
  },
]
`;

exports[`policy queries > return policies for user 1`] = `
[
  {
    "id": "5e6f5519ede3631176831f4c",
    "org_id": "5f2ab36c60d2294576efa646",
    "statements": [],
  },
]
`;

exports[`policy queries > return policies for user, filtering out those for locked orgs 1`] = `
[
  {
    "id": "5e6f5519ede3631176831f4c",
    "org_id": "5f2ab36c60d2294576efa646",
    "statements": [],
  },
  {
    "id": "5f3d63ce7a73979876d9c4c7",
    "statements": [],
  },
]
`;
